secondModuleSecondCoding/src/data_structures/DoublyCircularLinkedList.java

```diff
@@ -0,0 +1,96 @@
+package data_structures;
+
+public class DoublyCircularLinkedList<T> {
+
+    private Node<T> first;
+    private Node<T> last;
+    private Integer size;
+
+    public DoublyCircularLinkedList() {
+        this.first = null;
+        this.last = null;
+        this.size = 0;
+    }
+
+    public Node<T> getFirst() {
+        return first;
+    }
+
+    public void setFirst(Node<T> first) {
+        this.first = first;
+    }
+
+    public Node<T> getLast() {
+        return last;
+    }
+
+    public void setLast(Node<T> last) {
+        this.last = last;
+    }
+
+    public Integer getSize() {
+        return size;
+    }
+
+    public void setSize(Integer size) {
+        this.size = size;
+    }
+
+    public Node<T> add(T value) {
+        Node newNode = new Node<>(value);
```

Comment:
Here it will be a good idea to create this object, this way:
Node<T> newNode = new Node(value);
To prevent from getting warnings